server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java

```diff
@@ -981,6 +1021,7 @@ public VolumeVO resizeVolume(ResizeVolumeCmd cmd) throws ResourceAllocationExcep
                 newSize = newDiskOffering.getDiskSize();
             }
 
+            newSizeInGb = newSize >> 30;
```

Comment:
bit shift...

Have you checked if there is a library to convert between these bases? If there is none, what about we create an utils method to execute this job?